core/src/main/java/jenkins/util/InterceptingExecutorService.java

```diff
@@ -18,6 +18,7 @@
  * @author Kohsuke Kawaguchi
  * @since 1.557
  */
+@Deprecated
```

Comment:
`@Restricted(NoExternalUse.class) @RestrictedSince("TODO")`? Then plugins cannot trivially ignore a deprecation warning.

Reply:
Sure! That'll mean that jenkinsci/workflow-cps-plugin#448 needs to be merged before `workflow-cps` can bump its baseline to a core that contains this PR, but since jenkinsci/workflow-cps-plugin#448 has already been approved and it's likely going to be a while since `workflow-cps` bumps its baseline, the tradeoff seems reasonable.

Reply:
FWIW we can do both, `@Deprecated` for the IDE, `@Restricted` for the build :)

Reply:
I suppose so, but that doesn't seem to be the common idiom in this codebase. I think I'd rather stick with the prevailing usage for consistency, unless there are some compelling advantages to deviate in this specific case.

Reply:
Not a big deal either way, but most things annotated `@RestrictedSince` are `@Deprecated`. `@RestrictedSince` is generally used in two situations:

* Something was noticed afterwards as being unrestricted even though it should be restricted – often Jelly helper methods and such, that aren't considered stable API.
* We're deprecating something and want to make this _really_ noticeable in plugins and not just to plugin developers who review deprecated calls.

The latter case is present several times throughout the code base, from a quick GH search: https://github.com/jenkinsci/jenkins/blob/f6378676b11288cc2525852b0a1a441d147e31d3/core/src/main/java/hudson/model/queue/AbstractQueueSorterImpl.java#L42-L55 https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/src/main/java/hudson/util/AbstractTaskListener.java#L12-L14 https://github.com/jenkinsci/jenkins/blob/22fcc54bd3879c765b5595371e57fd180860daff/core/src/main/java/hudson/scm/SCMDescriptor.java#L82-L93 https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/src/main/java/hudson/slaves/DelegatingComputerLauncher.java#L97-L99 https://github.com/jenkinsci/jenkins/blob/22fcc54bd3879c765b5595371e57fd180860daff/core/src/main/java/hudson/org/apache/tools/tar/TarOutputStream.java#L47-L50 https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/src/main/java/hudson/org/apache/tools/tar/TarInputStream.java#L45-L47 https://github.com/jenkinsci/jenkins/blob/f6378676b11288cc2525852b0a1a441d147e31d3/core/src/main/java/hudson/util/TimeUnit2.java#L72-L74 https://github.com/jenkinsci/jenkins/blob/ca646b3742e6921d332078d0b287fd359970d7e9/core/src/main/java/hudson/model/Slave.java#L223-L237 https://github.com/jenkinsci/jenkins/blob/ca646b3742e6921d332078d0b287fd359970d7e9/core/src/main/java/hudson/Functions.java#L984-L997 https://github.com/jenkinsci/jenkins/blob/9f9ed7b60db1bc67210f1f4368f8ace218a5b9c0/core/src/main/java/jenkins/model/Jenkins.java#L4810-L4822 https://github.com/jenkinsci/jenkins/blob/7c821f4839ce308a56c3b0130ab3f6fe663ad37c/core/src/main/java/hudson/Util.java#L1593-L1595

I think this API is essentially the latter case, so both restricting and deprecating seems in line with what we've been doing for a while already.

Reply:
Good point. Done.